internal/server/event_handlers.go

```diff
@@ -451,6 +447,34 @@ func createNotifier(ctx context.Context, kubeClient client.Client, provider *api
 	return sender, token, nil
 }
 
+// getTLSConfigForProvider - retrieves the TLS configuration from the provider's certSecretRef or secretRef.
+func getTLSConfigForProvider(ctx context.Context, kubeClient client.Client, provider *apiv1beta3.Provider) (tlsConfig *tls.Config, err error) {
+	secretRef := types.NamespacedName{
+		Namespace: provider.GetNamespace(),
+	}
+	// certSecretRef takes precedence over secretRef as it is explicitly specified for TLS configuration
+	if provider.Spec.CertSecretRef != nil {
+		secretRef.Name = provider.Spec.CertSecretRef.Name
+		tlsConfig, err = secrets.TLSConfigFromSecretRef(ctx, kubeClient, secretRef)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get TLS config: %w", err)
+		}
+		return
+	}
+	// if certSecretRef is not specified, and if the provider is a git provider then
+	// attempt to get TLS config from secretRef if ca.crt exists in secretRef
+	if isGitProvider(provider.Spec.Type) {
+		if provider.Spec.SecretRef != nil {
+			secretRef.Name = provider.Spec.SecretRef.Name
+			tlsConfig, err = secrets.TLSConfigFromSecretRef(ctx, kubeClient, secretRef)
```

Comment:
This secret is already being fetched inside `extractAuthFromSecret()`, let's move the `.Get()` from there to `getTLSConfigForProvider()` and pass the `*corev1.Secret` down to both `extractAuthFromSecret()` and `getTLSConfigForProvider()`.